File: backend/src/models/SMENU.ts
import { Model, DataTypes } from 'sequelize';
import db from '../config/database';

class SMENU extends Model {
    public id!: any;
    public name!: any;
    public note!: any;
    public status!: any;
    public usermodifiedid!: any;
    public timemodified!: any;
    public timecreated!: any;
    public sortorder!: any;
    public usercreatedid!: any;
    public parentid!: any;
    public parentdir!: any;
    public itemtype!: any;
    public autoid!: any;
    public simageid!: any;
    public operation!: any;
    public stabledescid!: any;
    public sformid!: any;
    public bold!: any;
    public italic!: any;
    public image!: any;
    public toolbar!: any;
    public toolbarindex!: any;
    public shortcut!: any;
    public toolbartitlte!: any;
    public visibleconfigid!: any;
    public sreportid!: any;
    public viewcount!: any;
    public sfunctionid!: any;
    public loai!: any;
    public loctheoloai!: any;
    public hideinmenu!: any;
    public menucode!: any;
    public viewonts!: any;
    public viewonmobi!: any;
    public viewonweb!: any;
}

SMENU.init({
    ID: {
        type: DataTypes.INTEGER,
        primaryKey: true, autoIncrement: true,
    },
    NAME: {
        type: DataTypes.STRING,
        
    },
    NOTE: {
        type: DataTypes.STRING,
        
    },
    STATUS: {
        type: DataTypes.BOOLEAN,
        
    },
    USERMODIFIEDID: {
        type: DataTypes.INTEGER,
        
    },
    TIMEMODIFIED: {
        type: DataTypes.DATE,
        
    },
    TIMECREATED: {
        type: DataTypes.DATE,
        
    },
    SORTORDER: {
        type: DataTypes.INTEGER,
        
    },
    USERCREATEDID: {
        type: DataTypes.INTEGER,
        
    },
    PARENTID: {
        type: DataTypes.INTEGER,
        
    },
    PARENTDIR: {
        type: DataTypes.STRING,
        
    },
    ITEMTYPE: {
        type: DataTypes.STRING,
        
    },
    AUTOID: {
        type: DataTypes.INTEGER,
        
    },
    SIMAGEID: {
        type: DataTypes.INTEGER,
        
    },
    OPERATION: {
        type: DataTypes.STRING,
        
    },
    STABLEDESCID: {
        type: DataTypes.INTEGER,
        
    },
    SFORMID: {
        type: DataTypes.INTEGER,
        
    },
    BOLD: {
        type: DataTypes.BOOLEAN,
        
    },
    ITALIC: {
        type: DataTypes.BOOLEAN,
        
    },
    IMAGE: {
        type: DataTypes.BLOB,
        
    },
    TOOLBAR: {
        type: DataTypes.STRING,
        
    },
    TOOLBARINDEX: {
        type: DataTypes.STRING,
        
    },
    SHORTCUT: {
        type: DataTypes.STRING,
        
    },
    TOOLBARTITLTE: {
        type: DataTypes.STRING,
        
    },
    VISIBLECONFIGID: {
        type: DataTypes.INTEGER,
        
    },
    SREPORTID: {
        type: DataTypes.INTEGER,
        
    },
    VIEWCOUNT: {
        type: DataTypes.INTEGER,
        
    },
    SFUNCTIONID: {
        type: DataTypes.INTEGER,
        
    },
    LOAI: {
        type: DataTypes.STRING,
        
    },
    LOCTHEOLOAI: {
        type: DataTypes.STRING,
        
    },
    HIDEINMENU: {
        type: DataTypes.STRING,
        
    },
    MENUCODE: {
        type: DataTypes.STRING,
        
    },
    VIEWONTS: {
        type: DataTypes.STRING,
        
    },
    VIEWONMOBI: {
        type: DataTypes.STRING,
        
    },
    VIEWONWEB: {
        type: DataTypes.STRING,
        
    },
}, {
    sequelize: db.sequelize,
    modelName: 'SMENU',
    tableName: 'SMENU',
    timestamps: false
});

export default SMENU;

File: frontend/Models/SMENU.cs
using System;

namespace QuanLyBar.Client.Models
{
    public class SMENU
    {
        public int? Id { get; set; }
        public string Name { get; set; }
        public string Note { get; set; }
        public bool? Status { get; set; }
        public int? UsermodifiedId { get; set; }
        public DateTime? Timemodified { get; set; }
        public DateTime? Timecreated { get; set; }
        public int? Sortorder { get; set; }
        public int? UsercreatedId { get; set; }
        public int? ParentId { get; set; }
        public string Parentdir { get; set; }
        public string Itemtype { get; set; }
        public int? AutoId { get; set; }
        public int? SimageId { get; set; }
        public string Operation { get; set; }
        public int? StabledescId { get; set; }
        public int? SformId { get; set; }
        public bool? Bold { get; set; }
        public bool? Italic { get; set; }
        public byte[] Image { get; set; }
        public string Toolbar { get; set; }
        public string Toolbarindex { get; set; }
        public string Shortcut { get; set; }
        public string Toolbartitlte { get; set; }
        public int? VisibleconfigId { get; set; }
        public int? SreportId { get; set; }
        public int? Viewcount { get; set; }
        public int? SfunctionId { get; set; }
        public string Loai { get; set; }
        public string Loctheoloai { get; set; }
        public string Hideinmenu { get; set; }
        public string Menucode { get; set; }
        public string Viewonts { get; set; }
        public string Viewonmobi { get; set; }
        public string Viewonweb { get; set; }
    }
}
